cmd/probe/smartprobe.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"github.com/golang/glog"
+	"github.com/openebs/node-disk-manager/cmd/controller"
+	"github.com/openebs/node-disk-manager/pkg/smart"
+)
+
+// smartProbe contains required variables for populating diskInfo
+type smartProbe struct {
+	Controller      *controller.Controller
+	SmartIdentifier *smart.Identifier
+}
+
+const (
+	smartProbeName     = "smart probe"
+	smartProbeState    = defaultEnabled
+	smartProbePriority = 2
+)
+
+func init() {
+	go func() {
+		ctrl := <-controller.ControllerBroadcastChannel
+		if ctrl == nil {
+			glog.Error("unable to configure", smartProbeName)
+			return
+		}
+		var pi controller.ProbeInterface = &smartProbe{Controller: ctrl}
+		newPrgisterProbe := &registerProbe{
+			priority:       smartProbePriority,
+			probeName:      smartProbeName,
+			probeState:     smartProbeState,
+			probeInterface: pi,
+			controller:     ctrl,
+		}
+		newPrgisterProbe.register()
+	}()
+}
+
+// newSmartProbeForFillDiskDetails returns smartProbe struct which helps populate diskInfo struct.
+func newSmartProbeForFillDiskDetails(devPath string) *smartProbe {
```

Comment:
I really don't things we need to name these things SmartXXX That's all very confusing.

We should have disk objects, that have a member that represents SMART info, but we should not mix the terms here

Reply:
Sure @gila . 
It would be good to have a member in disk object that represent SMART info or info provided by smart but at the same time we are having this name since this file is dedicated to smart probe and the details given by it .
@gila Is it ok to have something like newSmartProbeToFillBasicDiskDetails() so that it can convey that it is being used to fill basic disk details not smart ones.